src/main/java/com/revature/studyforce/flashcard/controller/FlashcardController.java

```diff
@@ -1,100 +1,144 @@
+
     @Autowired
-    FlashcardRepo repo;
+    public FlashcardController(FlashcardService flashcardService){
+        this.flashcardService = flashcardService;
+    }
 
     /**
-     * getAll() method mapped to HTTP GET requests ("/all")
-     * @param page - number of offsets away from 0
-     * @param offset - number of Flashcards per page
-     * @return - returns Page of paginated Flashcards
+     * Retrieves a Page of flashcards {@link FlashcardService#getAll(int, int, String, String)}
+     * @param page - number of offsets away from 0 (defaults to 0)
+     * @param offset - number of elements per page [5|10|25|50|100] - defaults to 10
+     * @param sortBy - column to sort by ["difficulty"|"topic"|"created"|"resolved"] defaults to creator if sortby could not be understood
+     * @param order - order in which the Page is displayed ["ASC"|"DESC"]
+     * @return - returns a Page of Flashcards according the the given page, offset, sortBy, and order parameters
      */
-    @GetMapping("/all")
-    public @ResponseBody Page<Flashcard> getAll(
+    @GetMapping
+    public @ResponseBody Page<FlashcardAllDTO> getAll(
             @RequestParam(name = "page", defaultValue = "0", required = false) int page,
-            @RequestParam(name = "offset", defaultValue = "10", required = false) int offset) {
-        return repo.findAll(PageRequest.of(page, offset));
+            @RequestParam(name = "offset", defaultValue = "10", required = false) int offset,
+            @RequestParam(name = "sortby", defaultValue = "id", required = false) String sortBy,
+            @RequestParam(name = "order", defaultValue = "desc", required = false) String order) {
+        return flashcardService.getAll(page, offset, sortBy, order);
     }
 
     /**
-     * getAllByDifficulty() method mapped to HTTP GET requests ("/difficulty")
-     * @param page - number of offsets away from 0
-     * @param offset - number of Flashcards per offset
-     * @param difficulty - limits returned Flashcards to the given difficulty
-     * @return - returns a List of paginated Flashcards sorted by difficulty
+     * Retrieves a Page of flashcards with a given difficulty {@link FlashcardService#getAllByDifficulty(int, int, String, String, int)}
+     * @param page - number of offsets away from 0 (defaults to 0)
+     * @param offset - number of elements per page [5|10|25|50|100] - defaults to 10
+     * @param sortBy - column to sort by ["difficulty"|"topic"|"created"|"resolved"] defaults to creator if sortby could not be understood
+     * @param order - order in which the Page is displayed ["ASC"|"DESC"]
+     * @param difficulty - only return flashcards with the given difficulty
+     * @return - returns a Page of Flashcards according the the given page, offset, sortBy, order, and difficulty parameters
      */
     @GetMapping("/difficulty")
-    public @ResponseBody List<Flashcard> getAllByDifficulty(
+    public @ResponseBody Page<FlashcardAllDTO> getAllByDifficulty(
```

Comment:
Consider changing required RequestParam(/difficulty?difficulty=hard&...) to PathParam (/{hard}?...&...)

But I'm approving if this is all I find.